framework/include/functions/Function.h

```diff
@@ -87,6 +87,20 @@ class Function :
    */
   virtual RealGradient gradient(Real t, const Point & p);
 
+  /**
+   * Get the time derivative of the function
+   * \param t The time
+   * \param p The point in space (x,y,z)
+   * \return The time derivative of the function at the specified time and location
+   */
+  virtual Real timeDerivative(Real t, const Point & p);
+
+  /**
+    * Check to see whether function can calculate time derivative
+    * \return true if time derivative is defined, false otherwise
+    */
+  virtual inline bool haveTimeDerivative() { return false; }
```

Comment:
There is somewhat inconsistent use of ```have``` and ```has``` for such functions in MOOSE. 

Guys! Let's start a five page thread about that!!!!